CRT.c

```diff
@@ -793,12 +794,16 @@ static int CRT_colorSchemes[LAST_COLORSCHEME][LAST_COLORELEMENT] = {
    [COLORSCHEME_BROKENGRAY] = { 0 } // dynamically generated.
 };
 
+static bool retainScreenOnExit = false;
+
 int CRT_scrollHAmount = 5;
 
 int CRT_scrollWheelVAmount = 10;
 
 ColorScheme CRT_colorScheme = COLORSCHEME_DEFAULT;
 
+extern int iterationsRemaining;
```

Comment:
Another option might be to store it in the new struct Machine, alongside the other sample-time related fields there (e.g. realtime, realtimeMs, monotonicMs)?  That way its accessible in all the places that need it.

Reply:
@natoscott I'm not sure if `CRT.c` has access to the `Machine` structure (nor if it should). AFAICS this is currently not available there (and I'm not really convinced that we should change that).

Reply:
Based on what I have understood, the `Machine` structure is being used to keep track of machine-related stats such as memory usage and active CPUs.

> I'm not sure if CRT.c has access to the Machine structure

I concur. Currently, CRT.c doesn't seem to have access to the `Machine` struct but including Machine.h in CRT.c should fix that.

Reply:
@BenBE yeah, you're right - its definitely there where we need it in ScreenManager but not in Settings (and agree, its not the right thing to do adding it there).

@valdaarhun struct Machine has information about sampling too, which is what 'iterationsRemaining' is, so it could find a logical home there.

| static bool retainScreenOnExit = false;

This variable name for consistency with many other globals in CRT.c should be prefixed with CRT_ I guess.  I think the cleanest approach might be to have another flag passed into CRT_init, alongside allowUnicode, like 'retainScreen' to set the CRT.c global, and maintain the 'remainingIterations' counter within struct Machine alongside the sample timing fields there, for ScreenManager to update.  But, just my 2c... there's plenty of other ways too, I'm not strongly of any one opinion here.

Reply:
> struct Machine has information about sampling too, which is what 'iterationsRemaining' is, so it could find a logical home there.

Ah, ok. I have understood this now.

> I think the cleanest approach might be to have another flag passed into CRT_init, alongside allowUnicode, like 'retainScreen' to set the CRT.c global, and maintain the 'remainingIterations' counter within struct Machine alongside the sample timing fields there, for ScreenManager to update

I think this approach is quite neat.